apps/condo/domains/resident/utils/serverSchema/findOrganizationsByAddress.js

```diff
@@ -11,112 +10,206 @@ const {
+    
+    if (!receipts.length && checkAccountNumberUrl) {
+        const { status, services } = await getAccountsWithOnlineInteractionUrl(checkAccountNumberUrl, tin, accountNumber)
+        if (status === ONLINE_INTERACTION_CHECK_ACCOUNT_SUCCESS_STATUS) {
+            receipts = services.map(service => ({
+                category: service.category,
+                number: service.account.number,
+                routingNumber: service.bankAccount.routingNumber,
+                bankAccount: service.bankAccount.number,
+            }))
         }
-        return [organization.id, {
-            id: context.id,
-            period: get(context, 'lastReport.period'),
-            checkAccountNumberUrl: remoteInteractions[context.integration],
-        }]
-    }))
+    }
+
+    if (!receipts.length) receipts = null
+
+    const meters = await getOrganizationMeters(organization, addressKey, { accountNumber }, properties)
+
+    return {
+        id: organization.id,
+        name: organization.name,
+        tin: organization.tin,
+        type: organization.type,
+        receipts,
+        meters,
+    }
 }
 
-async function getOrganizationsWithMeters (organizationIds, query = {}) {
-    const organizationsWithMeters = await Promise.all(organizationIds.map(async id => {
-        const [meter] = await itemsQuery('Meter', {
-            where: { organization: { id }, deletedAt: null, ...query },
-            first: 1,
+async function getOrganizationReceipts (billingContext, addressKey, query = {}) {
+    if (!billingContext) return []
+    const billingAccounts = await find('BillingAccount', {
+        context: { id: billingContext.id },
+        property: { addressKey, deletedAt: null },
+        deletedAt: null,
+        ...query,
+    })
+    let receipts = []
+
+    if (billingAccounts.length && get(billingContext, 'lastReport.period')) {
+        receipts = await find('BillingReceipt', {
+            context: { id: billingContext.id },
+            property: { addressKey, deletedAt: null },
+            deletedAt: null,
+            period: billingContext.lastReport.period,
+            account: { id_in: billingAccounts.map(({ id }) => id) },
         })
-        return meter ? id : null
-    }))
-    return new Set(organizationsWithMeters.filter(Boolean))
+
+        const billingAccountsNumbersIndex = Object.fromEntries(
+            billingAccounts.map(account => [account.id, account.number])
+        )
+        receipts = receipts.map(receipt => ({
+            category: receipt.category,
+            balance: receipt.toPay,
+            accountNumber: billingAccountsNumbersIndex[receipt.account],
+            routingNumber: receipt.recipient.bic,
+            bankAccount: receipt.recipient.bankAccount,
+            address: get(receipt, 'raw.address', null),
+        }))
+    }
+
+    return receipts
+}
+
+async function getOrganizationMeters (organization, addressKey, query = {}, properties) {
+    let meters = await find('Meter', {
```

Comment:
Corrupted logic detected.

Organization can have meters on property, but it should not be shown to user, since its not resource owner on his house